### ISSUE TITLE
Document tradeof/difference between smallvec

### ISSUE THREAD

**Issue by lijiaqigreat** [Lokathor/tinyvec#168]
tl;dr:
It's really useful for people to decide which crate to choose. Could you add something to README?

Based on my research, I summarize it as:

They both solve the same issue with similar approach.
smallvec is older hence more stable and has more user. But it's owned be servo team, and is potentially dead.
tinyvec is unsafe-free (which I think should not be a hard rule, since bug-free != unsafe-free), and is still being actively improved.

In terms of time/op performance, tinyvec is still improving on this, the latest status is that it's faster for common ops and slower in other ops.

Other advantage for tinyvec:
more variants (such as ArrayVec)

Other advantage for smallvec:
smallvec has more feature flags which allow granular performance tweak.
Such as: union feature allows smaller mem size (32 byte to 24 byte).

I can open a separate issue if you want, but I think smallvec union feature is really nice, and would be nice for tinyvec to have it. But  it's impossible without unsafe code.

Could you add something like this to the doc?
I also want to know, is tinyvec trying to be smallvec but better? or tinyvec tries to solve for slightly different scenario?

If you want I can reword and submit a PR to add what I wrote to README.

**Comment by Lokathor** [comment]
I want to start by saying this: For the `tinyvec` crate, the `forbid(unsafe)` rule is absolute. I don't think all code needs to stick to such a rule (I started `bytemuck` after all), but this crate has self-imposed that `forbid(unsafe)` is part of our advertised contract with our users.

As to the main request: Yeah I could update the readme with a little compare and contrast segment.
